### ISSUE TITLE
DataFrameInternal uses deprecated class Matrix

### ISSUE THREAD

**Issue by khinsen** [PolyMathOrg/DataFrame#44]
The `Matrix` class has been deprecated in Pharo 7. Better use `Array2D` ?

**Comment by Ducasse** [comment]
Array2d and matrix are not the same. In fact long time ago I made the mistake to rename Array2d into matrix but it was missing many of the matrix operation. 

Stef

**Comment by khinsen** [comment]
Ahhh... the matrix-array debate. I was part of the team that designed NumPy in 1994, and we already had the debate back then. And it comes up again and again. In my opinion, there are two workable approaches:

 - The APL way: there are only arrays. Matrix operations are expressed in terms of arrays, requiring some adaptation for people coming from a maths-matrix background.
 - Arrays and matrices as separate classes, which may share implementation details but are not substitutable (in particular, none is a subclass of the other).

The second approach is probably easier to get used to for most people, but ends up requiring tons of conversions. My personal preference is the array-only approach.

Are there any plans to work in these issues in Pharo? For now, all I see in my Pharo 7 image is that `Matrix` is an empty and deprecated subclass of `Array2D`.

**Comment by Ducasse** [comment]
Array2d is an array 2d :) so we can understand what it is. Now if someone define on array2d or array2d like class the functions required for manipulating things like a matrix then why not :)
But in Matrix we got a Matrix with an Array2D api so I think that it was the worse case scenario and it was my mistake because back then nobody needed the Matrix/vector API. 
In Besset code he needed it and created it.
Now Pharo is one thing and the libraries another. Ideally we want in Pharo the least amount of classes and that the user can load what it wants. This is why I started to packaged, clean,... some lost collections that were not used in the image without tests too on my Containers-* projects. 
So I encourage you to push and gets young guys to build nice libraries. This is what we are doing with Oleksandr.
